src/Components/src/Microsoft.AspNetCore.Components/Rendering/HtmlRenderer.cs

```diff
@@ -73,6 +73,42 @@ private IEnumerable<string> RenderComponent(Type componentType, ParameterCollect
             }
         }
 
+        /// <summary>
+        /// Renders a component into a sequence of <see cref="string"/> fragments that represent the textual representation
+        /// of the HTML produced by the component.
```

Comment:
No, it's because producing an IEnumerable<string> is the simplest thing we can do right now. We can change this in the future to do something different and faster, but I want to leave it like this for now.

MVC wise, some level of the stack will have to buffer given that IHtmlContent doesn't support async.